compiler/rustc_mir_build/src/thir/pattern/_match.rs

```diff
@@ -832,6 +832,139 @@ impl Slice {
     fn arity(self) -> u64 {
         self.pattern_kind().arity()
     }
+
+    /// The exhaustiveness-checking paper does not include any details on
+    /// checking variable-length slice patterns. However, they are matched
+    /// by an infinite collection of fixed-length array patterns.
+    ///
+    /// Checking the infinite set directly would take an infinite amount
+    /// of time. However, it turns out that for each finite set of
+    /// patterns `P`, all sufficiently large array lengths are equivalent:
+    ///
+    /// Each slice `s` with a "sufficiently-large" length `l ≥ L` that applies
+    /// to exactly the subset `Pₜ` of `P` can be transformed to a slice
+    /// `sₘ` for each sufficiently-large length `m` that applies to exactly
+    /// the same subset of `P`.
+    ///
+    /// Because of that, each witness for reachability-checking from one
+    /// of the sufficiently-large lengths can be transformed to an
+    /// equally-valid witness from any other length, so we only have
+    /// to check slice lengths from the "minimal sufficiently-large length"
+    /// and below.
```

Comment:
```suggestion
    /// Because of that, each witness for reachability-checking of one
    /// of the sufficiently-large lengths can be transformed to an
    /// equally-valid witness of any other length, so we only have
    /// to check slices of the "minimal sufficiently-large length"
    /// and less.
```